src/gnatss/ops/io.py

```
@@ -184,7 +182,7 @@ def gather_files_all_procs(
             # if we are loading solution from cache
             continue
 
-        if proc == "main" or hasattr(config, proc):
+        if hasattr(config, proc):
             all_files_dict.update(gather_files(config, proc, mode))
     return all_files_dict
```

For the sake of non silent failure, maybe raise error above this like if all_files_dict is empty

---

I added a ValueError message in case all_files_dict{}, see lines 188-190 of ops/io.py. However, after doing some testing I could not get this error to trigger when running an example in a Jupyter notebook. I tried various flavors of commenting out input data fields in the config yaml and all of my permutation failed at the data_loading() stage upstream of gather_files_all_procs().

While this may be a good thing overall from the perspective of future-proofing, I can't say with certainty whether this will ever come into play (although I guess there is precedent).